crates/sui-graphql-rpc/src/context_data/pg_backend.rs

```diff
@@ -42,6 +42,12 @@ impl GenericQueryBuilder<Pg> for PgQueryBuilder {
         }
         query
     }
+    fn get_obj_by_type(object_type: String) -> objects::BoxedQuery<'static, Pg> {
+        objects::dsl::objects
+            .filter(objects::dsl::object_type.eq(object_type))
+            .limit(1)
```

Comment:
The `limit(1)` is kind of funky here, and definitely warrants a doc comment explaining that it's intended for use with singleton types or something.